usl_pipeline/study_area_uploader/study_area_uploader/chunkers/study_area_chunkers.py

```diff
@@ -0,0 +1,180 @@
+        chunk_size: Size of chunks in cells (over both axes).
+        extend_soil_classes_chunk_border: Indicates that chunk areas for soil classes
+            should be extended by chunk_size in all directions around each chunk.
+        input_elevation_band: Band number in input GeoTIFF elevation file that elevation
+            data should be read from.
+        default_nodata_value: NODATA value that should be used to store output elevation
+            chunks.
+    """
+    logging.info("Preparing elevation chunks...")
+    elevation_chunks_dir = work_dir / "elevation_chunks"
+    elevation_chunks_dir.mkdir(parents=True, exist_ok=True)
+    elevation_chunk_descriptors = elevation_chunkers.split_geotiff_into_chunks(
+        input_data.elevation_file_path, chunk_size, elevation_chunks_dir
+    )
+    with open(input_data.elevation_file_path, "rb") as input_fd:
+        header = elevation_readers.read_from_geotiff(input_fd, header_only=True).header
+
+    # Chunking all present polygon layers
+    boundaries_chunks_dir = _chunk_polygons_if_present(
+        header, chunk_size, input_data.boundaries_polygons, work_dir, "boundaries"
+    )
+    buildings_chunks_dir = _chunk_polygons_if_present(
+        header, chunk_size, input_data.buildings_polygons, work_dir, "buildings"
+    )
+    green_areas_chunks_dir = _chunk_polygons_if_present(
+        header, chunk_size, input_data.green_areas_polygons, work_dir, "green_areas"
+    )
+    soil_classes_chunk_additional_border_cells = (
+        chunk_size if extend_soil_classes_chunk_border else 0
+    )
+    soil_classes_chunks_dir = _chunk_polygons_if_present(
+        header,
+        chunk_size,
+        input_data.soil_classes_polygons,
+        work_dir,
+        "soil_classes",
+        support_mask_values=True,
+        chunk_additional_border_cells=soil_classes_chunk_additional_border_cells,
+    )
+
+    for elevation_chunk_descriptor in elevation_chunk_descriptors:
+        y_chunk_index = elevation_chunk_descriptor.y_chunk_index
+        x_chunk_index = elevation_chunk_descriptor.x_chunk_index
+        chunk_file_name = f"chunk_{y_chunk_index}_{x_chunk_index}"
+        logging.info("Exporting %s...", chunk_file_name)
+        elevation_chunk_file_path = elevation_chunk_descriptor.path
+        # Let's standardize the band number (to 1) and the NODATA_value (to default one)
+        with open(elevation_chunk_file_path, "rb") as input_file:
+            chunk_elevation = elevation_readers.read_from_geotiff(
+                input_file,
+                band=input_elevation_band,
+                no_data_value=default_nodata_value,
+            )
+        if (
+            chunk_elevation.header.row_count < chunk_size
+            or chunk_elevation.header.col_count < chunk_size
+        ):
+            y_pad = chunk_size - chunk_elevation.header.row_count
+            x_pad = chunk_size - chunk_elevation.header.col_count
+            chunk_elevation.data = numpy.pad(
```

Comment:
Is this logic that should be part of `elevation_chunkers.split_geotiff_into_chunks` itself?

Reply:
I thought about it but then decided that we never read elevation GeoTiff directly from python inside `elevation_chunkers.split_geotiff_into_chunks`, we do `GDAL.translate` to cut chunks, nothing else. On the contrary, in the uploader script I'm reading chunk elevation data anyway to repack the band number and default NODATA. So I'm optimizing the number of read/writes with this one. Let me know if you would want to move the padding deeper into elevation chunker.

Reply:
sounds good -- might just leave a comment with some thoughts.

Reply:
Done. Also I swapped X and Y axis indices in chunk naming to match it to your schema.